vulture_os/toolkit/api_parser/harfanglab/harfanglab.py

```diff
@@ -217,11 +225,25 @@ def execute(self):
                 logger.debug(f"[{__parser__}]:execute: {msg}", extra={'frontend': str(self.frontend)})
 
                 if total == 0:
-                    # Means that there are no logs available. It may be for two
-                    # reasons: no log during this period or logs not available at
-                    # request time.
-                    # If there are no logs, no need to write them and we should not
-                    # set the last_api_call.
+                    """
+                    Means that there are no logs available. It may be for two
+                    reasons: no log during this period or logs not available at
+                    request time. If there are no logs, no need to write them.
+                    """
+                    break
+
+                if total >= 10000:
+                    """
+                    API have a return treshold setted to 10k logs by default (maybe elastic underlying layer)
+                    for every server response. This leads to an infinte loop in case of a large volume returned
+                    on the requested time's range (mostly ~0.001eps but spikes can be much larger)
```

Comment:
```suggestion
                    API have a return threshold set to 10k logs by default (maybe elastic underlying layer)
                    for every server response. This leads to an infinite loop in case of a large volume returned
                    on the requested time range (mostly ~0.001eps but spikes can be much larger)
```